crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3276,7 +3295,9 @@ impl<'db> TypeInferenceBuilder<'db> {
                     "Relative module resolution `{}` failed: too many leading dots",
                     format_import_from_module(*level, module),
                 );
-                report_unresolved_module(&self.context, module_ref, *level, module);
+                if is_import_reachable() {
+                    report_unresolved_module(&self.context, module_ref, *level, module);
+                }
```

Comment:
We could also move the `is_import_reachable` check inside `report_unresolved_module`, but that would require passing many parameters (or the closure). Not sure what's better.

Reply:
Or we could get rid of `report_unresolved_module` and replace it with a method on the `TypeInferenceBuilder`?

```rs
impl TypeInferenceBuilder {
    fn report_unresolved_module(&self, range: impl Ranged, level: u32, module: Option<&str>) {
        let file_scope_id = self.scope().file_scope_id(self.db());

        if !self.index
            .is_node_reachable(self.db(), file_scope_id, NodeKey::from_node(import_from))
        {
            return;
        }

        self.context.report_lint(
            &UNRESOLVED_IMPORT,
            range,
            format_args!(
                "Cannot resolve import `{}{}`",
                ".".repeat(level as usize),
                module.unwrap_or_default()
            ),
        );
    }
}

Reply:
(could also just have the method accept `NodeKey` and `TextRange` to avoid monomorphisation)

Reply:
Thanks!